aalondon/static/js/components/MeetingForm.js

```diff
@@ -0,0 +1,445 @@
+          return false
+        } else if (value.length === 0) {
+          return false
+        }
+        else {
+          return true
+        }
+
+      }),
+
+    submission: Yup.string()
+      .required('Submission type is required'),
+
+    startTime: Yup.string()
+      .required('Start time is required')
+      .test("is-valid", "Start time needs to be in 24 hour format e.g. 13:30", function (value) {
+        return moment(value, "HH:mm", true).isValid();
+      }),
+
+
+
+    email: Yup.string()
+      .email('Email is invalid')
+      .required('Email is required'),
+    description: Yup.string()
+      .required('Description is required'),
+
+  };
+
+  let validationPhysicalShape =
+  {
+    postcode: Yup.string()
+      .required('Postcode is required'),
+    address: Yup.string()
+      .required('Address is required'),
+  }
+
+  let validationOnlineShape = {
+    link: Yup.string()
+      .required('Link is required')
+      .url('Please enter valid url!'),
+
+  }
+
+  let finalValidationShape
+  if (formType === 'Hybrid') {
+    finalValidationShape = { ...validationAllShape, ...validationPhysicalShape, ...validationOnlineShape }
+  } else if (formType === 'faceToFace') {
+    finalValidationShape = { ...validationAllShape, ...validationPhysicalShape }
+  } else {
+    finalValidationShape = { ...validationAllShape, ...validationOnlineShape }
+  }
+  const validationSchema = Yup.object().shape(finalValidationShape);
+
+  return (
+    <Formik
+      initialValues={{
+        formType: formType,
+        day: '',
+        submission: '',
+        intergroup: '',
+        startTime: '',
+        link: '',
+        password: '',
+        paymentLink: '',
+        address: '',
+        postcode: '',
+        whatThreeWords: '',
+        email: '',
+        description: '',
+        notes: '',
+        wheelchair: false,
+        signed: false,
+        lgbt: false,
+        chits: false,
+        childFriendly: false,
+        outdoors: false,
+        creche: false,
+        temporaryClosure: false
+      }}
+      validationSchema={validationSchema}
+      onSubmit={fields => {
+
+
+        alert('SUCCESS!! :-)\n\n' + JSON.stringify(fields, null, 4))
+
+
+      }}>
+      {({ errors, status, touched }) => (
+        <Form>
+
+          <div className="form-group">
+            <label htmlFor="day">Day</label>
+            <SemanticField
+              name="day"
+              component={Dropdown}
+              options={dayOptions}
+              multiple
+              selection
+              placeholder="Please select day of week"
+              id={"day"}
+              value={[]}
+              className={'form-control' + (errors.day && touched.day ? ' is-invalid' : '')}
+
+            />
+
+            <ErrorMessage name="day" component="div" className="invalid-feedback" />
+          </div>
+          <div className="form-group">
+            <label htmlFor="submission">Submission</label>
```

Comment:
`<label ...>Is this a new entry to AALondon, or an update to an existing entry?</label>`